smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/fault/DeserializerWrapper.java

```diff
@@ -137,6 +139,13 @@ private T tryToRecover(String topic, Headers headers, byte[] data, Exception exc
             }
         } else {
             KafkaLogging.log.unableToDeserializeMessage(topic, exception);
+            if (failOnDeserializationErrorWithoutHandler) {
+                source.reportFailure(exception, true);
+                if (exception instanceof KafkaException) {
+                    throw (KafkaException) exception;
+                }
+                throw new KafkaException(exception);
+            }
```

Comment:
Javadoc of this method needs an update :-)

Reply:
Fixed.